src/js/playlist/playlist.js

```diff
@@ -27,6 +27,9 @@ define([
             // include selected file in item for backwards compatibility
             item.file = item.sources[0].file;
 
+            // set preload for the item
+            item.preload = item.preload || preload;
```

Comment:
This works, can be even better though! 

             if (originalSource.drm || itemDrm) {
                 originalSource.drm = originalSource.drm || itemDrm;
             }